pwiz_tools/Skyline/Controls/Graphs/GraphFullScan.cs

```diff
@@ -734,17 +749,37 @@ private void  CreateSingleScan()
                     var curveItem = _graphHelper.GraphControl.AddGraphItem(GraphPane, item, false);
                     curveItem.Label.IsVisible = false;
                 }
+            }
+            // Create curve for all points to provide shading behind stick graph.
+            if (_msDataFileScanHelper.MsDataSpectra.Length > 0 && !_msDataFileScanHelper.MsDataSpectra[0].Centroided)
+            {
+                var item = new SpectrumShadeItem(allPointList, Color.FromArgb(100, 225, 225, 150), @"all");
+                var curveItem = _graphHelper.GraphControl.AddGraphItem(GraphPane, item, false);
+                curveItem.Label.IsVisible = false;
+            }
+            GraphPane.SetScale(CreateGraphics());
 
-                // Create curve for all points to provide shading behind stick graph.
-                if (_msDataFileScanHelper.MsDataSpectra.Length > 0 && !_msDataFileScanHelper.MsDataSpectra[0].Centroided)
+            if (Settings.Default.ShowFullScanMassError)
+            {
+                massErrors = new double[_msDataFileScanHelper.ScanProvider.Transitions.Length];
+
+                //create and initialize a map of transition,accumulator pairs
+                var meanErrorsMap =
+                    new Dictionary<Identity, IntensityAccumulator>(_msDataFileScanHelper.ScanProvider.Transitions.Length);

```

Comment:
The "Equals" method on Identity usually does not do what you want.
You should either use "IdentityEqualityComparer" when you construct this Dictionary, or you should use "IdentityPath" as the key in this dictionary.

Reply:
You mean like this: 
`                var meanErrorsMap =
                    new Dictionary<Identity, IntensityAccumulator>(_msDataFileScanHelper.ScanProvider.Transitions.Length, new IdentityEqualityComparer<Identity>());
`